src/manage_sql_secinfo.c

```diff
@@ -2081,14 +2242,435 @@ insert_scap_cpe_details (inserts_t *inserts, element_t cpe_item)
+                                  quoted_name,
+                                  quoted_deprecated_by_id);
+
+          deprecated_by_inserts->current_chunk_size++;
+
+          g_free (quoted_deprecated_by_id);
+        }
+    }
+
+  quoted_cpe_name_id = sql_quote (cpe_name_id);
+  quoted_title = sql_quote (title_text ? title_text : "");
+
+  first = inserts_check_size (inserts);
+
+  g_string_append_printf (inserts->statement,
+                          "%s ('%s', '%s', '%s', %li, %li, %d, '%s')",
+                          first ? "" : ",",
+                          quoted_name,
+                          quoted_name,
+                          quoted_title,
+                          modification_time,
+                          modification_time,
+                          deprecated,
+                          quoted_cpe_name_id);
+
+  inserts->current_chunk_size++;
+
+  g_free (quoted_title);
+  g_free (quoted_name);
+  g_free (quoted_cpe_name_id);
+
+  return 0;
+}
+
+/**
+ * @brief Insert a SCAP CPE from JSON.
+ *
+ * @param[in]  inserts            Pointer to SQL buffer.
+ * @param[in]  product_item       JSON object from the products list.
+ *
+ * @return 0 success, -1 error.
+ */
+static int
+handle_json_cpe_refs (inserts_t *inserts, cJSON *product_item)
+{
+  cJSON *cpe_item, *refs, *refs_item;
+  gchar *name, *quoted_name;
+
+  assert (inserts);
+
+  cpe_item = cJSON_GetObjectItemCaseSensitive (product_item, "cpe");
+  if (! cJSON_IsObject (cpe_item))
+    {
+      g_warning ("%s: 'cpe' field in product missing or not an object",
+                 __func__);
+      return -1;
+    }
+
+  name = json_object_item_string (cpe_item, "cpeName");
+  if (name == NULL)
+    {
+      g_warning ("%s: 'cpeName' field missing or not a string", __func__);
+      return -1;
+    }
+
+  refs = cJSON_GetObjectItemCaseSensitive (cpe_item, "refs");
+  if (! cJSON_IsArray (refs))
+    {
+      g_debug ("%s: 'refs' field missing or not an array", __func__);
+      return 0;
+    }
+
+  quoted_name = fs_to_uri_convert_and_quote_cpe_name (name);
+  cJSON_ArrayForEach (refs_item, refs)
+    {
+      int first;
+      char *ref, *type;
+      gchar *quoted_ref, *quoted_type;
+      ref = json_object_item_string (refs_item, "ref");
+      if (ref == NULL)
+        {
+          g_warning ("%s: 'ref' field missing or not a string", __func__);
+          g_free (quoted_name);
+          return -1;
+        }
+      type = json_object_item_string (refs_item, "type");
+      quoted_ref = sql_quote (ref ? ref : "");
+      quoted_type = sql_quote (type ? type : "");
+      
+      first = inserts_check_size (inserts);
+
+      g_string_append_printf (inserts->statement,
+                              "%s ('%s', '%s', '%s')",
+                              first ? "" : ",",
+                              quoted_name,
+                              quoted_ref,
+                              quoted_type);
+
+      inserts->current_chunk_size++;
```

Comment:
You may want to free `quoted_ref` and `quoted_type`
```suggestion
      inserts->current_chunk_size++;
      g_free (quoted_ref);
      g_free (quoted_type);      
```